### ISSUE TITLE
Failed to load new campaign level

### ISSUE THREAD

**Issue by Xardion495** [Regalis11/Barotrauma#9048]
- [x] I have searched the issue tracker to check if the issue has already been reported.
- Perhaps complements the issue #9023 but my case not related to mods

**Description**
If the arrival point is dead end there is no way to complete successes missions. Even if come back to start outpost and dock, nothing happen.
But if restart the round and dock to outpost immediately (before completing missions) the round ends automatically, so you can processing the campaign, but missions considered as failed.

**Version**
0.17.15.0 (ReleaseWindows, branch release, revision 62bb4ecab3)

**Additional information**
`Failed to load a new campaign level. No available level transitions (transition type: ProgressToNextEmptyLocation, current location: Milda, selected location: Apep, leaving sub: Winterhalter, at start: False, at end: True)`

![scr](https://user-images.githubusercontent.com/17722887/166330100-5f06edad-0d7f-4b0b-85c1-7169987f06f6.png)

Docked but the round no ends
![scr1e](https://user-images.githubusercontent.com/17722887/166330511-4700a340-b4fb-435d-b6fa-1233ba3e9fa5.png)

**Comment by Regalis11** [comment]
I'm afraid I could not reproduce this. :/ Tested in a map with the seed "1" and I was able to load into a new level at both ends of the level.
![image](https://user-images.githubusercontent.com/14351070/166418866-2d66ba9a-703d-4280-bfda-07a6e0ffdde7.png)

If you still have this campaign save, could you attach it here?

**Comment by Xardion495** [comment]
Looking ahead I suspect that the problem consists in Pirate Sub mission which require to end round exactly at the arrival point. Guess this nuance applies to all other missions with the same requirement.

Save attached. The sub docked in Apep outpost. The mods warnings can be ignored cause they didn't contain non-vanilla items or critical important structure generates to processing campaign.
[save.zip](https://github.com/Regalis11/Barotrauma/files/8610360/save.zip)

What exactly we did step-by-step:

1. Took the pirate sub mission. Went to the next round
2. Explored the level. Faced with pirates
3. Destroyed the pirate crew, received a success pirate mission message
4. Arrived at the end point (Milda), chose Apep, clicked 'Start'
5. Nothing happened + received error message in log
6. Went back to the start point, docked to Apep, nothing happened + no error/warn message in log
7. Announced: 'guys, we screwed', restarted the round
8. Immediately docked to Apep after level beginning. Went to the next round

Looks like the game has a standalone handler for level toggling which is stopped after throwing an error.
